src/Duration.php

```diff
@@ -109,6 +109,11 @@ public static function create($duration): self
         }
 
         $duration = (string) $duration;
+
+        if (strpos($duration, 'P') === 0) {
```

Comment:
I would still go with a regular expression 
```php
<?php

$regexp = '/^P
    (?:(?:\d+Y)?(?:\d+M)?(?:\d+D)?)?
    (?:T(?:\d+H)?(?:\d+M)?(?:\d+S)?)?
 $/xi';
```

Reply:
Thanks for the fast review! I just updated the PR accordingly!